### PR TITLE
SSD support

### PR DESCRIPTION
**Pull request by Rhymmor**
#85  
In this PR:
1. Two packages with Caffe SSD libraries (CPU and GPU with CUDA)
- `lib-caffe-ssd-cpu` 
- `lib-caffe-ssd-cuda`

2. Corresponding programs for SSD detection (only with default mode)
- `caffe-detection-ssd`
- `caffe-detection-ssd-cuda`

3. COCO-2014, VOC-2007, VOC-2012 datasets
- `dataset-coco-2014`
- `datataset-voc-2007`
- `dataset-voc-2012`

4. Pretrained SSD 300 and 512 models (trained on COCO and VOC)
- `caffemodel-ssd-coco-300`
- `caffemodel-ssd-coco-512`
- `caffemodel-ssd-voc-300`
- `caffemodel-ssd-voc-512`

### CODE REVIEW

File: package/lib-caffe-ssd-cpu/.cm/meta.json [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        "BLAS_TYPE": "Open",
+        "BUILD_MATLAB": 0,
+        "BUILD_PYTHON": 0,
+        "BUILD_PYTHON_LAYER": 0,
+        "CMAKE_BUILD_SHARED_LIBS": 0,
+        "CMAKE_CONFIG": "Release",
+        "CPU_ONLY": 1,
+        "PACKAGE_BUILD_TYPE": "cmake",
+        "PACKAGE_CONFIGURE_FLAGS": "",
+        "PACKAGE_CONFIGURE_FLAGS_ANDROID": "-DBUILD_python=OFF -DBUILD_docs=OFF -DUSE_HDF5=OFF",
```
Comment (DVEfremov) [inline package/lib-caffe-ssd-cpu/.cm/meta.json:15]: some package unified some are not
but looks like there is no problem to unify all packages

File: package/caffemodel-ssd-voc-300/.cm/meta.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }, 
+    "skip_file_check": "yes", 
+    "version": "trunk" 
+  }, 
+  "end_full_path": { 
+    "linux": "VGG_VOC0712_SSD_300x300_iter_120000.caffemodel", 
+    "win": "VGG_VOC0712_SSD_300x300_iter_120000.caffemodel" 
+  }, 
+  "only_for_host_os_tags": [ 
+    "windows",
```
Comment (DVEfremov) [inline package/caffemodel-ssd-voc-300/.cm/meta.json:38]: can't see bat file (only install.sh)
have you  tried it run on windows ?

**Review by DVEfremov — CHANGES_REQUESTED** [review]
please tack a look and le'ts discus verbally

File: package/caffemodel-ssd-voc-300/.cm/meta.json (hunk 2 above)
Comment (Rhymmor) [reply package/caffemodel-ssd-voc-300/.cm/meta.json:38]: No, only for linux right now

Comment (dsavenko) [reply package/caffemodel-ssd-voc-300/.cm/meta.json:38]: In this case it shouldn't mention 'win' and 'windows' in the meta.

**Comment by dsavenko** [comment]
Will provide an updated PR soon.

**Pull request closed without merging**